automation/jenkins-package-feed.sh

```diff
@@ -0,0 +1,240 @@
+
+# process script arguments
+args_number="$#"
+while [[ $# -ge 1 ]]; do
+	arg=$1
+	case $arg in
+		-h|--help)
+			print_help
+			exit 0
+			;;
+		-m|--machine)
+			if [ -z "$2" ]; then
+				echo "-m|--machine argument needs a machine name"
+				exit 1
+			fi
+			MACHINE="$2"
+			;;
+		--shared-dir)
+			if [ -z "$2" ]; then
+				echo "--shared-dir needs directory name where to store shared downloads and sstate data"
+				exit 1
+			fi
+			JENKINS_PERSISTENT_WORKDIR="$2"
+			shift
+			;;
+		-a|--additional-variable)
+			if [ -z "$2" ]; then
+				echo "\"$1\" needs an argument in the format VARIABLE=VALUE"
+				exit 1
+			fi
+			if echo "$2" | grep -vq '^[A-Za-z0-9_-]*='; then
+				echo "\"$2\" has the wrong argument format for \"$1\". Read help."
+				exit 1
+			fi
+			BARYS_ARGUMENTS_VAR="$BARYS_ARGUMENTS_VAR $1 $2"
+			shift
+			;;
+		-b|--build-flavor)
+			if [ -z "$2" ]; then
+				echo "-b|--build-flavor argument needs a build type"
+				exit 1
+			fi
+			buildFlavor="${buildFlavor:-$2}"
+			;;
+		--meta-balena-branch)
+			if [ -z "$2" ]; then
+				echo "--meta-balena-branch argument needs a meta-balena branch name (if this option is not used, the default value is __ignore__)"
+				exit 1
+			fi
+			metaResinBranch="${metaResinBranch:-$2}"
+			;;
+		--supervisor-tag)
+			if [ -z "$2" ]; then
+				echo "--supervisor-tag argument needs a resin supervisor tag name (if this option is not used, the default value is __ignore__)"
+				exit 1
+			fi
+			supervisorTag="${supervisorTag:-$2}"
+			;;
+		--esr)
+			ESR="true"
+			;;
+		--preserve-build)
+			BARYS_ARGUMENTS_VAR=${BARYS_ARGUMENTS_VAR//--remove-build/}
+			;;
+		--preserve-container)
+			REMOVE_CONTAINER=""
+			;;
+		-t|--build-target)
+			if [ -z "$2" ]; then
+				echo "-t|--build-target argument needs a target image name"
+				exit 1
+			fi
+			BARYS_ARGUMENTS_VAR="$BARYS_ARGUMENTS_VAR --bitbake-target $2"
+			shift
+			;;
+	esac
+	shift
+done
+
+JENKINS_DL_DIR=$JENKINS_PERSISTENT_WORKDIR/shared-downloads
+JENKINS_SSTATE_DIR=$JENKINS_PERSISTENT_WORKDIR/$MACHINE/sstate
+metaResinBranch=${metaResinBranch:-__ignore__}
+supervisorTag=${supervisorTag:-__ignore__}
+
+# Sanity checks
+if [ -z "$MACHINE" ] || [ -z "$JENKINS_PERSISTENT_WORKDIR" ] || [ -z "$buildFlavor" ]; then
+	echo -e "\n[ERROR] You are missing one of these arguments:\n
+\t -m <MACHINE>\n
+\t --shared-dir <PERSISTENT_WORKDIR>\n
+\t --build-flavor <BUILD_FLAVOR_TYPE>\n\n
+Run with -h or --help for a complete list of arguments.\n"
+	exit 1
+fi
+
+if [ "$buildFlavor" = "dev" ]; then
+	BARYS_ARGUMENTS_VAR="$BARYS_ARGUMENTS_VAR --development-image"
+	DEVELOPMENT_IMAGE=yes
+elif [ "$buildFlavor" = "prod" ]; then
+	DEVELOPMENT_IMAGE=no
+else
+	echo "[ERROR] No such build flavor: $buildFlavor."
+	exit 1
+fi
+
+# When supervisorTag is provided, you the appropiate barys argument
+if [ "$supervisorTag" != "__ignore__" ]; then
+	BARYS_ARGUMENTS_VAR="$BARYS_ARGUMENTS_VAR --supervisor-tag $supervisorTag"
+fi
+
+# Checkout meta-balena
```

Comment:
This can probably be removed also, are we envisioning we'll built from other branches than master?